applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_FIC_element.cpp

```diff
@@ -144,7 +144,10 @@ void UPwSmallStrainFICElement<TDim, TNumNodes>::InitializeNonLinearIteration(con
         Variables.B = b_matrices[GPoint];
 
         // Compute infinitessimal strain
-        this->CalculateStrain(Variables, GPoint);
+        Variables.F            = this->CalculateDeformationGradient(GPoint);
+        Variables.detF         = MathUtils<>::Det(Variables.F);
```

Comment:
Indeed we add a bit of extra duplication in this PR, it's a bit of a trade-off with limiting the use of the ElementVariables container. Since the Variables.F is used in the strain vector calculation and passed on to the constitutive parameters, I think it's better to keep it explicit like it is now. 

In a next PR, which is now still in draft status, lists are extracted for F, detF and the strain vectors (see https://github.com/KratosMultiphysics/Kratos/pull/12363/files#diff-5a457c1c0c5d61a1ff97989e636c83f79506ea59254d16659303b48efde6f3cbR318-R333). I think that state is acceptable, especially since it becomes clearer and clearer who uses what (instead of everyone grabbing anything out of the ElementVariables bucket). I'm curious to know your opinion on this!

As for the question of when do/don't we use SetConstitutiveParameters: I think that depends on what we are actually calculating on the integration points. E.g. when just calculating the StrainVector, we don't need to use the constitutive law. However, if we want to convert that strain to a stress vector and generate that as an output, we do need to set the constitutive parameters.

Reply:
Hi Richard, thank you for the info. My message is not clear, probably, because it was done in the late evening. ;) My comment was on moving Variables.detF         = MathUtils<>::Det(Variables.F); to SetConstitutiveParameters . It is clear that everything will change soon. 
Concerning the next PR, it seems these functions are the same on the first glance. If it is true then they should be in utilities.